addons/links/src/preview.ts

```diff
@@ -14,23 +21,24 @@ const generateUrl = id => {
   )}`;
 };
 
-const valueOrCall = args => value => (typeof value === 'function' ? value(...args) : value);
+const valueOrCall = (args: string[]) => (value: string | ((...args: string[]) => string)) =>
+  typeof value === 'function' ? value(...args) : value;
 
-export const linkTo = (kind, story) => (...args) => {
+export const linkTo = (kind: string, story?: string) => (...args: string[]) => {
   const resolver = valueOrCall(args);
   navigate({
     kind: resolver(kind),
     story: resolver(story),
   });
 };
 
-export const hrefTo = (kind, name) =>
+export const hrefTo = (kind: string, name: string): Promise<string> =>
   new Promise(resolve => {
     resolve(generateUrl(toId(kind, name)));
   });
 
-const linksListener = e => {
-  const { sbKind: kind, sbStory: story } = e.target.dataset;
+const linksListener = (e: SyntheticEvent<HTMLLinkElement>) => {
+  const { sbKind: kind, sbStory: story } = e.currentTarget.dataset;
```

Comment:
`currentTarget` is always `document` which makes it throw on each click, see e.g.

https://next--storybooks-html.netlify.com/?path=/story/welcome--welcome